src/libcore/slice/memchr.rs

```diff
@@ -102,8 +102,24 @@ pub fn memrchr(x: u8, text: &[u8]) -> Option<usize> {
     let ptr = text.as_ptr();
     let usize_bytes = mem::size_of::<usize>();
 
+    // a version of align_offset that says how much must be *subtracted*
+    // from a pointer to be aligned.
+    #[inline(always)]
+    fn align_offset_down(ptr: *const u8, align: usize) -> usize {
```

Comment:
Consider `slice::align_to::<usize>()`? I believe that does exactly what a large part of this code does.

Reply:
Hm, plausible. Should I then also do that for `memchr`?

Reply:
Actually it's not so easy, we want to do the main loop on a `u8` slice, not `usize` (or actually pairs of `usize`, which is the steps that the loop is taking).